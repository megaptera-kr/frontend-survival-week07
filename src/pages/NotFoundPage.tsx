import { useNavigate } from 'react-router-dom';

export default function NotFoundPage() {
  const navigate = useNavigate();

  const handlePreviousPage = () => {
    navigate(-1);
  };

  return (
    <section>
      <h1>Not Found Page</h1>
      <button type="button" onClick={handlePreviousPage}>
        Previous
      </button>
    </section>
  );
}
