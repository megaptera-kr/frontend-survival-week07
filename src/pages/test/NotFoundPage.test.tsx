import { render } from '@testing-library/react';
import { Route } from 'react-router-dom';

import { withRouter } from '../../tests/helper';
import NotFoundPage from '../NotFoundPage';

describe('Not Found Page', () => {
  it('render correctly', () => {
    const { asFragment } = render(withRouter(<Route path="/" element={<NotFoundPage />} />));

    expect(asFragment()).toMatchSnapshot();
  });
});
