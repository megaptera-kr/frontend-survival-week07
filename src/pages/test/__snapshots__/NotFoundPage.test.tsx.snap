// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Page render correctly 1`] = `
<DocumentFragment>
  <section>
    <h1>
      Not Found Page
    </h1>
    <button
      type="button"
    >
      Previous
    </button>
  </section>
</DocumentFragment>
`;
